### PR TITLE
Nexus: small fixes in pseudopotential methods

### PR DESCRIPTION
**Pull request by jtkrogel**
Fixes to pseudopotential.py: add colors for high angular channels, update contourf usage, fix read/write edge case for QMCPACK pseudopotential file I/O.

Also catch a small bug in new-ish warning code: a full Exception entered as `message` in `__nexus_showwarning`.

### CODE REVIEW

**Comment by jtkrogel** [comment]
Ready now

File: nexus/nexus/generic.py [hunk 1]
```diff
@@ -85,7 +85,7 @@ def __nexus_showwarning(message, category, filename, lineno, file=None, line=Non
         indent = message.indent
         cls    = f"{message.cls}:" if message.cls is not None else ""
     else:
-        msg = message
+        msg = str(message)
```
Comment (brockdyer03) [inline nexus/nexus/generic.py:88]: I feel like I noticed a similar problem a little bit ago, but couldn't figure out where it came from. I'd be curious to know if you have an example of that happening.

File: nexus/nexus/pseudopotential.py [hunk 2]
```diff
@@ -1646,7 +1656,7 @@ def write_qmcpack(self,filepath=None):
                 if n%3==0:
                     L2 += dpad
                 #end if
-                L2 += '{0:22.14e}'.format(d)
+                L2 += ' {0:22.14e}'.format(d)
```
Comment (brockdyer03) [inline nexus/nexus/pseudopotential.py:1659]: I think this might be a good case for an f-string.
```python
L2 += f' {d:22.14e}'
```

File: nexus/nexus/pseudopotential.py [hunk 3]
```diff
@@ -1668,7 +1678,7 @@ def write_qmcpack(self,filepath=None):
                     if n%3==0:
                         semilocal+=dpad
                     #end if
-                    semilocal+='{0:22.14e}'.format(d)
+                    semilocal+=' {0:22.14e}'.format(d)
```
Comment (brockdyer03) [inline nexus/nexus/pseudopotential.py:1681]: Same here

File: nexus/nexus/pseudopotential.py [hunk 4]
```diff
@@ -2762,6 +2772,11 @@ def read(self,filepath,format=None):
             if l2.format!='r*V':
                 self.error('unrecognized potential format: {0}\nthe only supported format is r*V'.format(l2.format))
             #end if
+            if isinstance(l2.radfunc.data,str):
+                # fix edge case: no spaces between written floats
```
Comment (brockdyer03) [inline nexus/nexus/pseudopotential.py:2776]: Any chance you can add a test for this edge case?

**Review by brockdyer03 — CHANGES_REQUESTED** [review]
The only thing that is a "mandatory" change is adding a test to make sure the edge case handling works as expected. I'd like to move to f-strings, but that's style, and the error thing is curiosity.

File: nexus/nexus/pseudopotential.py (hunk 4 above)
Comment (jtkrogel) [reply nexus/nexus/pseudopotential.py:2776]: Notice the spaces introduced on the "format" lines.  This prevents the numbers from running together (the "edge case").  The updated read just handles previously written files that somehow managed to saturate the format string.

File: nexus/nexus/generic.py (hunk 1 above)
Comment (jtkrogel) [reply nexus/nexus/generic.py:88]: I don't remember the exact context that tripped it.

File: nexus/nexus/pseudopotential.py (hunk 4 above)
Comment (jtkrogel) [reply nexus/nexus/pseudopotential.py:2776]: Test added

**Comment by jtkrogel** [comment]
Addressed gating request.  Ready for merge.

**Review by brockdyer03 — APPROVED** [review]
LGTM

**Comment by prckent** [comment]
Test this please
